src/main/scala/com/spark3d/geometry/CubeEnvelope.scala

```diff
@@ -0,0 +1,625 @@
+      min(p1.y, min(p2.y, p3.y)), max(p1.y, max(p2.y, p3.y)),
+      min(p1.z, min(p2.z, p3.z)), max(p1.z, max(p2.z, p3.z))
+    )
+  }
+
+  /**
+    * Creates an cube Envelope for a region defined by two coordinates.
+    *
+    * @param p1 first coordinate
+    * @param p2 second coordinate
+    */
+  def this(p1: Point3D, p2: Point3D) {
+    this(
+      min(p1.x, p2.x), max(p1.x, p2.x),
+      min(p1.y, p2.y), max(p1.y, p2.y),
+      min(p1.z, p2.z), max(p1.z, p2.z)
+    )
+  }
+
+  /**
+    * Creates an cube Envelope for a region defined one coordinate. The cube Envelope in this case will be a point.
+    *
+    * @param p1 the coordinate
+    */
+  def this(p1: Point3D) {
+    this(p1.x, p1.x, p1.y, p1.y, p1.z, p1.z)
+  }
+
+  /**
+    * Clones an existing cube Envelope to create a duplicate cube Envelope.
+    *
+    * @param env original cube Envelope to be cloned
+    */
+  def this(env: CubeEnvelope) {
+    this(env.minX, env.maxX, env.minY, env.maxY, env.minZ, env.maxZ)
+  }
+
+  /**
+    * Checks if this is a null envelope or not.
+    *
+    * @return if this cube Envelope is null (empty Geometry) or nots
+    */
+  def isNull(): Boolean = {
+    ((minX > maxX) || (minY > maxY) || (minZ > maxZ))
+  }
+
+
+  /**
+    * Sets this cube Envelope to null
+    *
+    */
+  def setToNull: Unit = {
+    minX = 0
+    maxX = -1
+    minY = 0
+    maxY = -1
+    minZ = 0
+    maxZ = -1
+  }
+
+  /**
+    * Return the difference between max and min X values of the cube Envelope.
+    *
+    * @return maxX - minX, or 0 if the cube Envelope is null
+    */
+  def getXLength(): Double = {
+
+    if (isNull) {
+      return 0.0
+    }
+    maxX - minX
+  }
+
+  /**
+    * Returns the difference between max and min Y value of the cube Envelope.
+    *
+    * @return maxX - minY, or 0 if the cube Envelope is null
+    */
+  def getYLength(): Double = {
+
+    if (isNull) {
+      return 0.0
+    }
+    maxY - minY
+  }
+
+  /**
+    * Return the difference between max and min Z value of the cube Envelope.
+    *
+    * @return maxZ - minZ, or 0 if the cube Envelope is null
+    */
+  def getZLength(): Double = {
+
+    if (isNull) {
+      return 0.0
+    }
+    maxZ - minZ
+  }
+
+  /**
+    * Gets minimum extent of this cube Envelope across all three dimensions.
+    *
+    * @return the minimum extent of this cube Envelope
+    */
+  def minExtent(): Double = {
+    if (isNull) {
+      return 0.0
+    }
+
+    min(getXLength(), min(getYLength(), getZLength()))
+  }
+
+  /**
+    * Gets maximum extent of this cube Envelope across all three dimensions.
+    *
+    * @return the maximum extent of this cube Envelope
+    */
+  def maxExtent(): Double = {
+    if (isNull) {
+      return 0.0
+    }
+
+    max(getXLength(), max(getYLength(), getZLength()))
+  }
+
+  /**
+    * Returns the area of the cube Envelope.
+    *
+    * @return the area od the envelope, 0.0 if the cube Envelope is null
+    */
+  def getArea(): Double = {
```

Comment:
I do not think this is right.
If you mean area, then this is the sum of the area of each side.
If you mean volume, then you have the right formule (lenght * height * breadth).

In any case, change the name or the formule ;-)